### PR TITLE
[Enhancement] Refactor and deprecate Invoice repository class

### PR DESCRIPTION
**Pull request by Prometee**
The actual `Sylius\InvoicingPlugin\Repository\DoctrineInvoiceRepository` has been deprecated and a new class has been defined in the `src/Doctrine/ORM` folder with a name following the entity name.
The corresponding interface has been also deprecated and the new one is owning the only required method.

I propose this PR because the actual repository was not correctly used : it was an override of `sylius_invoicing_plugin.custom_repository.invoice` instead of a default value in the resource plugin configuration.

### CODE REVIEW

**Comment by Prometee** [comment]
@pamil @GSadee I fix the ECS error, but this fix should maybe part of another PR I guess :

```
[ERROR] Class "Symplify\PackageBuilder\Console\HelpfulApplicationTrait" not found while loading        
         "Symplify\EasyCodingStandard\Console\EasyCodingStandardConsoleApplication".  
```

File: src/Repository/InvoiceRepository.php [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 namespace Sylius\InvoicingPlugin\Repository;
 
+@trigger_error('The "InvoiceRepository" interface is deprecated since version 0.11.1 Use standardized interface located at "src/Doctrine/ORM/" instead.');
```
Comment (GSadee) [inline src/Repository/InvoiceRepository.php:7]: As it is not a stable version of the plugin yet, I would be for removing deprecation and just refactor the approach to `InvoiceRepository` as you did in this PR.

**Review by GSadee — COMMENTED** [review]
I've extracted your commit with fix to a separate PR #190 to make the build green again 😃

File: src/Repository/InvoiceRepository.php (hunk 1 above)
Comment (Prometee) [reply src/Repository/InvoiceRepository.php]: Done and I add a note the `UPGRADE.md`.

File: UPGRADE.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+### UPGRADE FROM 0.11.0 TO 0.11.1
```
Comment (GSadee) [inline UPGRADE.md:1]:
```suggestion
### UPGRADE FROM 0.11.0 TO 0.12.0
```

File: UPGRADE.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+
+1. The custom repository has been removed :
+
+  - removed the repository class `DoctrineInvoiceRepository` and replaced by `Sylius\InvoicingPlugin\Doctrine\ORM\InvoiceRepository`
+  - removed the related service `sylius_invoicing_plugin.custom_repository.invoice` use `sylius_invoicing_plugin.repository.invoice` instead
```
Comment (GSadee) [inline UPGRADE.md:6]:
```suggestion
  - removed the related service `sylius_invoicing_plugin.custom_repository.invoice`, use `sylius_invoicing_plugin.repository.invoice` instead
```

File: UPGRADE.md [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
+  - removed the related interface `InvoiceRepository` use `Sylius\InvoicingPlugin\Doctrine\ORM\InvoiceRepositoryInterface` instead
```
Comment (GSadee) [inline UPGRADE.md:7]:
```suggestion
  - removed the related interface `Sylius\InvoicingPlugin\Repository\InvoiceRepository`, use `Sylius\InvoicingPlugin\Doctrine\ORM\InvoiceRepositoryInterface` instead
```

File: UPGRADE.md [hunk 5]
```diff
@@ -1,3 +1,11 @@
+### UPGRADE FROM 0.11.0 TO 0.11.1
+
+1. The custom repository has been removed :
+
+  - removed the repository class `DoctrineInvoiceRepository` and replaced by `Sylius\InvoicingPlugin\Doctrine\ORM\InvoiceRepository`
```
Comment (GSadee) [inline UPGRADE.md:5]:
```suggestion
  - removed the repository class `Sylius\InvoicingPlugin\Repository\DoctrineInvoiceRepository` and replaced by `Sylius\InvoicingPlugin\Doctrine\ORM\InvoiceRepository`
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Francis! :tada:

**Pull request merged**